test/parallel/test-tls-multi-key.js

```diff
@@ -2,52 +2,162 @@
 const common = require('../common');
 const assert = require('assert');
 
+// Test multi-identity ('key')/multi-algorithm scenarios.
+
 if (!common.hasCrypto) {
   common.skip('missing crypto');
   return;
 }
 const tls = require('tls');
 const fs = require('fs');
 
-const options = {
+// Key and cert are ordered as ec, rsa.
+test({
   key: [
     fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
   ],
   cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
-    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem')
-  ]
-};
-
-const ciphers = [];
-
-const server = tls.createServer(options, function(conn) {
-  conn.end('ok');
-}).listen(0, function() {
-  const ecdsa = tls.connect(this.address().port, {
-    ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
-    rejectUnauthorized: false
-  }, function() {
-    ciphers.push(ecdsa.getCipher());
+  ],
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert are in opposite order from each other (order is not required).
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+  ],
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key, cert, and pfx options can be used simultaneously.
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+  ],
+  pfx: fs.readFileSync(common.fixturesDir + '/keys/agent1.pfx'),
+  passphrase: 'sample',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs
+test({
```

Comment:
Looking at the cert chain of these tests. They are
```
ec10(agent10):ECC -> ca6:ECC -> ca5(selfsigned):ECC
agent10(agent10):RSA -> ca4:RSA -> ca2(selfsigned):RSA
```
so this cannot formed right cert chain. 
I think we should separate intermidiate cert and root cert as below. 
```diff
@@ -72,6 +76,10 @@ test({
   rsaCN: 'agent10.example.com',
   eccCN: 'agent10.example.com',
   ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ca4-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca6-cert.pem'),
+  ],
+  rootca: [
     fs.readFileSync(common.fixturesDir + '/keys/ca2-cert.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/ca5-cert.pem'),
   ],
@@ -118,7 +126,7 @@ test({
 function test(options) {
   const rsaCN = options.rsaCN || 'agent1';
   const eccCN = options.eccCN || 'agent2';
-  const ca = options.ca;
+  const rootca = options.rootca;
```